edi-module/component_group_reader.bal

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+isolated function readComponentGroup(string compositeText, EDISchema ediSchema, EDIFieldSchema fieldSchema)
+            returns EDIComponentGroup|Error? {
+    if compositeText.trim().length() == 0 {
+        return ();
+    }
+
+    string[] components = split(compositeText, ediSchema.delimiters.component);
+    if fieldSchema.truncatable {
+        int minFields = getMinimumCompositeFields(fieldSchema);
+        if components.length() < minFields {
+            return error Error(string `Composite mapping's field count does not match minimum field count of the truncatable field: ${fieldSchema.tag}.
+                Required minimum field count: ${minFields}. Found ${components.length()} fields. 
+                Composite mapping: ${fieldSchema.toJsonString()} | Composite text: ${compositeText}`);
+        }
+    } else if fieldSchema.components.length() != components.length() {
+        return error Error(string `Composite mapping's component count does not match field: ${fieldSchema.tag}. 
+                Composite mapping: ${fieldSchema.toJsonString()} | Composite text: ${compositeText}`);
+    }
```

Comment:
Simiilar to an approach in [comment](https://github.com/ballerina-platform/edi-tools/pull/2/files#r1186255541), see whether this validation also extracted to a single function. It seems same logic is repeating in multiple places like component_group_reader.bal, segment_reader.bal, subcomponent_group_reader.bal etc.

Reply:
It is not straight forward to do this as those functions work with three different schema types (although there are similarities among schemas).